android/shared/src/main/kotlin/kmp/android/shared/di/Module.kt

```diff
@@ -0,0 +1,23 @@
+package kmp.android.shared.di
+
+import android.content.Context
+import com.google.android.gms.location.LocationServices
+import kmp.android.shared.ErrorMessageProviderImpl
+import kmp.android.shared.device.LocationControllerImpl
+import kmp.android.shared.domain.controller.LocationController
+import kmp.android.shared.domain.usecase.GetLocationFlowUseCase
+import kmp.android.shared.domain.usecase.GetLocationFlowUseCaseImpl
+import kmp.shared.base.error.ErrorMessageProvider
+import org.koin.dsl.module
+
+val androidSharedModule = module {
+    single<ErrorMessageProvider> { ErrorMessageProviderImpl(get()) }
```

Comment:
Nit.: There is a nice way of defining classes with simple params (when everything is get())
`singleOf(::ErrorMessageProviderImpl) binds ErrorMessageProvider::class`
This way when there are added/removed parameters there is no need to change this definition 🙂 
(This is just FYI, no need to change it now)

Reply:
I tried to change it like you suggested and wasn't able to solve the errors it created quickly, so if it is a quick fix for you, feel free to add it to this PR, but otherwise, I would create a new task for it (I guess it could be in more places so it would take more time...)😁

Reply:
Usually it doesnt import the class automatically when using this, when I write it I first write the name of the Impl class I want to define and import it, then write the `singleOf(::` in front of it 😄 But don't worry about it in this PR, this is just to let you know it exists 🙂